src/components/EditProfileForm/index.jsx

```diff
@@ -0,0 +1,214 @@
+import React, { useState } from "react";
+import { InsertPhoto } from "@mui/icons-material";
+import { useDispatch, useSelector } from "react-redux";
+
+import {
+    editAuthUser,
+	editModalVisibility,
+	editUserProfile,
+	getAuthState,
+	getUserProfileState,
+} from "features";
+import { CircleProgressBar } from "components";
+import { useToast } from "hooks";
+
+const EditProfileForm = () => {
+	const dispatch = useDispatch();
+	const { authToken } = useSelector(getAuthState);
+	const { userProfile, userProfileLoading } =
+		useSelector(getUserProfileState);
+	const { showToast } = useToast();
+
+	const [userFormData, setUserFormData] = useState({ ...userProfile });
+	const { username, firstName, lastName, website, profileImage, bio } =
+		userFormData;
+
+	const handleUserFormDataChange = ({ target: { name, value } }) =>
+		setUserFormData((prevUserFormData) => ({
+			...prevUserFormData,
+			[name]: value,
+		}));
+
+	const handleOnProfileImageChange = (event) => {
+		const imageFile = event.target.files[0];
+		if (Math.floor(imageFile / 1000000) > 3) {
+			showToast("Image file size should be less than 3MB", "error");
+			return;
+		}
+		const url =
+			"https://api.cloudinary.com/v1_1/" +
+			process.env.REACT_APP_CLOUD_NAME +
+			"/image/upload";
+
+		const formData = new FormData();
+		formData.append("file", imageFile);
+		formData.append("upload_preset", "l4ggled4");
```

Comment:
put this in `.env` and then read, else anyone will be able to upload images in your cloud

Reply:
Yes, thank you. This will be reflected in the next PR